### PR TITLE
Chore: Ensure severity gets validated in tests with position data

### PR DESCRIPTION
**Pull request by antross**
## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/webhintio/hint)
- [x] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- [ ] Added/Updated related documentation.
- [ ] Added/Updated related tests.

## Short description of the change(s)

Likely superseded by #3297, but wanted to call this out separately in case we want to either (1) merge this fix first or (2) ensure this fix is covered by that PR.

Basically the way the checks were structured in `validateResults` allowed `severity` to be ignored if a `position` property was also part of the expected report as the block for `position` would return whether or not there was a match. Moving the `severity` check earlier fixes that issue.

### CODE REVIEW

**Comment by molant** [comment]
🤦‍♂️

We probably want to merge this right after the flattening of utils. Don't know when I'll have the time to update the other PR

**Comment by molant** [comment]
Just synced with @antross and we agreed this should be merged asap so we will merge this now.
